syllabus/curricula/plr/task_sampler.py

```diff
@@ -363,44 +401,32 @@ def sample(self, strategy=None):
                 if np.random.rand() > self.nu or not proportion_seen < 1.0:
                     return self._sample_replay_level()
 
-            # Otherwise, evaluate a new level
+            # Otherwise, sample a new level
             if self.robust_plr:
                 self.update_with_episode_data(self._evaluate_unseen_level())
```

Comment:
You call `self.update_with_episode_data` here and within `self._evaluate_unseen_level`. You should only keep one of them.
```suggestion
                    self._evaluate_unseen_level()
```